flight/Libraries/math/statistics.c

```diff
@@ -0,0 +1,352 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsMath Tau Labs math support libraries
+ * @{
+ * @addtogroup TauLabsStatistics Tau Labs statistics math support libraries
+ * @{
+ *
+ * @file       statistics.c
+ * @author     Kenn Sebesta, Copyright (C) 2015.
+ * @brief      Statistics support
+ *
+ *****************************************************************************/
+
+#include "statistics.h"
+
+#include <math.h>
+
+
+/**
+ * @brief bayes_filter Implements a bayesian testing under the effect of an
+ * actuator that can modify what is being sensed.
+ * Belief_H0 is the H0 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle has landed.
+ * Belief_H1 is the H1 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle is still flying.
+ *
+ * Inspired by Probablistic Robotics, Chap. 2.4, "Bayes Filters"
+ * @param belief_H0 The H0 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle has landed.
+ * @param belief_H1 The H1 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle is still flying.
+ * @param p_sense_condition_H0 The probability of sensing the H0 condition, given the real condition
+ * @param p_sense_condition_H1 The probability of sensing the H1 condition, given the real condition
+ * @param p_H0_Ux_H0 The probability of state H0 staying H0 after the control is applied
+ * @param p_H0_Ux_H1 The probability of state H1 becoming H0 after the control is applied
+ * @param p_H1_Ux_H0 The probability of state H0 becoming H1 after the control is applied
+ * @param p_H1_Ux_H1 The probability of state H1 staying H1 after the control is applied
+ */
+void bayes_filter(double *belief_H0, double *belief_H1,
```

Comment:
Well, the example is something we definitely need for various things-- enhanced failsafe and "air mode" need to have a good estimate of whether we're in flight.  Which could be heuristic or could be based on hypothesis filtering.

And, I don't know.  For many things, 2-3 9's is great, and float is good for 6+.  Especially in many circumstances we'll have a great initial prior ("almost certainly not flying"), fairly high certainty observations along the way (high throttle command?  sustained non-unity acceleration?  varying altimeter?)  and thresholds using the test are likely to be somewhere like 90-95%.